### ISSUE TITLE
Add transaction support

### ISSUE THREAD

**Issue by reyesml** [reyesml/RMT#13]
https://gorm.io/docs/transactions.html

Possibly get rid of "repos" pattern and lift db interactions into the interactors? 🤔

Another option is to have each repo method accept db as a param. Not sure how much I like this though. Originally, I wanted the interactors to be largely unaware of the underlying database mechanism.  Keeping db queries isolated to Repos limits the blast radius if I decide to swap out the ORM for something else in the future.  I'd like to maintain this isolation if possible.

Basically, I need a mechanism for hot-swapping the Repos' `db` attribute once a transaction has begun.
